realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -182,25 +174,30 @@ public static synchronized SyncSession getSession(SyncConfiguration syncConfigur
             throw new IllegalArgumentException("A non-empty 'syncConfiguration' is required.");
         }
 
-        if (SessionStore.hasSession(syncConfiguration)) {
-            return SessionStore.getPublicSession(syncConfiguration);
-        } else {
-            ObjectServerSession internalSession = new ObjectServerSession(
-                    syncConfiguration,
-                    authServer,
-                    syncConfiguration.getUser().getSyncUser(),
-                    syncConfiguration.getSyncPolicy(),
-                    syncConfiguration.getErrorHandler()
-            );
-            SyncSession publicSession = new SyncSession(internalSession);
-            SessionStore.addSession(publicSession, internalSession);
-            syncConfiguration.getUser().getSyncUser().addSession(publicSession);
-            syncConfiguration.getSyncPolicy().onSessionCreated(internalSession);
-            return publicSession;
+        SyncSession session = sessions.get(syncConfiguration.getPath());
+        if (session == null) {
+            session = new SyncSession(syncConfiguration);
+            sessions.put(syncConfiguration.getPath(), session);
         }
+
+        return session;
     }
 
-    public static AuthenticationServer getAuthServer() {
+    /**
+     * Remove the wrapped Java session.
+     * @param syncConfiguration configuration object for the synchronized Realm.
+     */
+    public static synchronized void removeSession(SyncConfiguration syncConfiguration) {
```

Comment:
unfortunately yes, this is called from `SyncObjectServerFacade#realmClosed` which is an internal package